cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/EndPointField.java

```diff
@@ -69,4 +70,19 @@ public String toString() {
       ", field='" + field + '\'' +
       '}';
   }
+
+  /**
+   * Checks for validity of an EndPointField.
+   *
+   * If in a pipeline a field is dropped, the source EndPointField corresponding to
+   * the dropped field maps to an empty EndPointField of the form
+   * `EndPointField{endPoint=EndPoint{namespace='null', name='null', properties='{}'}, field='null'}`.
+   * This method can be used to scan for such EndPointFields.
+   *
+   * @return true if an EndPointField is valid, false otherwise
+   */
+  public boolean isValid() {
```

Comment:
the grouping here is...odd. Why is there a parenthesis if all methods are using the AND operation?

Also, please make this a multiline string so it's easier to read.

```

Styling with Markdown is supported
    return endPoint != null 
    && endPoint.getName() != null 
    && endPoint.getNamespace() != null 
    && !endPoint.getProperties().isEmpty() 
    && field != null;
      ``

Reply:
I added the grouping to kind of indicate that the first set of checks is for the first field `EndPoint` and the nested fields and the second check is for the second field in `EndPointField` but yes, it isn't really required. Removed the grouping